packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -855,9 +866,15 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       );
     }
     let next = beginWork(current, workInProgress, nextRenderExpirationTime);
-
     if (__DEV__) {
       ReactDebugCurrentFiber.resetCurrentFiber();
+      if (isReplayingFailedUnitOfWork) {
+        // Currently replaying a failed unit of work. This should be unreachable,
+        // because the render phase is meant to be idempotent, and it should
+        // have thrown again. Since it didn't, rethrow the original error, so
+        // React's internal stack is not misaligned.
+        throw originalReplayError;
```

Comment:
Should we maybe clear the variable just before throwing so we don't hold onto it until next replay?

Reply:
Yeah good call